prov/verbs/src/verbs_ep.c

```diff
@@ -1441,6 +1441,23 @@ static int vrb_pep_bind(fid_t fid, struct fid *bfid, uint64_t flags)
 		if (ret)
 			return -errno;
 	}
+
+	if (vrb_gl_data.log_async_events) {
+		ret = fi_fd_nonblock(pep->id->verbs->async_fd);
+		if (ret)
+			goto async_err;
+		ret = ofi_epoll_add(pep->eq->epollfd, pep->id->verbs->async_fd,
+				    OFI_EPOLL_IN, &pep->pep_fid.fid);
+		if (ret) {
+async_err:
+			VRB_WARN(FI_LOG_DOMAIN, "Failed to enable async events for %s: %s\n",
+				 pep->info->domain_attr->name, strerror(ret));
+			pep->log_async_events = false;
+		} else {
+			pep->log_async_events = true;
+		}
+	}
```

Comment:
If the app wants blocking support, just add the async_fd to the epollfd, so that waiting on the EQ will be triggered if any async event occurs.  Logging should just always be enabled if blocking support is desired.  If the EQ doesn't have blocking support, there's nothing to do here.  See below.